rocketpy/rocket/aero_surface.py

```diff
@@ -1889,3 +1888,210 @@ def all_info(self):
+              Mach number, and return the drag coefficient. This function allows
+              for dynamic computation based on deployment and Mach number.
+            - If an array, it should be a 2D array with three columns: the first
+              column for deployment level, the second for Mach number, and the
+              third for the corresponding drag coefficient.
+            - If a string, it should be the path to a .csv or .txt file. The
+              file must contain three columns: the first for deployment level,
+              the second for Mach number, and the third for the drag
+              coefficient.
+            - If a Function, it must take two parameters: deployment level and
+              Mach number, and return the drag coefficient.
+
+            .. note:: For ``override_rocket_drag = False``, at
+                deployment level 0, the drag coefficient is assumed to be 0,
+                independent of the input drag coefficient curve. This means that
+                the simulation always considers that at a deployment level of 0,
+                the air brakes are completely retracted and do not contribute to
+                the drag of the rocket.
+
+        reference_area : int, float
+            Reference area used to calculate the drag force of the air brakes
+            from the drag coefficient curve. Units of m^2.
+        clamp : bool, optional
+            If True, the simulation will clamp the deployment level to 0 or 1 if
+            the deployment level is out of bounds. If False, the simulation will
+            not clamp the deployment level and will instead raise a warning if
+            the deployment level is out of bounds. Default is True.
+        override_rocket_drag : bool, optional
+            If False, the air brakes drag coefficient will be added to the
+            rocket's power off drag coefficient curve. If True, during the
+            simulation, the rocket's power off drag will be ignored and the air
+            brakes drag coefficient will be used for the entire rocket instead.
+            Default is False.
+        deployment_level : float, optional
+            Current deployment level, ranging from 0 to 1. Deployment level is the
+            fraction of the total airbrake area that is Deployment. Default is 0.
+        name : str, optional
+            Name of the air brakes. Default is "AirBrakes".
+
+        Returns
+        -------
+        None
+        """
+        super().__init__(name)
+        self.drag_coefficient_curve = drag_coefficient_curve
+        self.drag_coefficient = Function(
+            drag_coefficient_curve,
+            inputs=["Deployment Level", "Mach"],
+            outputs="Drag Coefficient",
+        )
+        self.reference_area = reference_area
+        self.clamp = clamp
+        self.override_rocket_drag = override_rocket_drag
+        self._deployment_level = deployment_level
```

Comment:
If this `deployment_level` is an initial value, I would run it through the standard property setter, as `self.deployment_level = deployment_level`.

This way out of bound initial inputs are clamped if this behavior was selected.

Reply:
Good catch